packages/mdc-textfield/README.md

```diff
@@ -344,19 +385,31 @@ complicated.
+| notifyTrailingIconAction() => void | Broadcasts a custom event "MDCTextfield:trailing-icon" denoting a user has clicked the trailing icon |
+| addClassToBottomLine(className: string) => void | Adds a class to the bottom line element |
+| removeClassFromBottomLine(className: string) => void | Removes a class from the bottom line element |
 | addClassToHelptext(className: string) => void | Adds a class to the help text element. Note that in our code we check for whether or not we have a help text element and if we don't, we simply return. |
 | removeClassFromHelptext(className: string) => void | Removes a class from the help text element. |
 | helptextHasClass(className: string) => boolean | Returns whether or not the help text element contains the current class |
-| setHelptextAttr(name: string, value: string) => void | Sets an attribute on the help text element |
-| removeHelptextAttr(name: string) => void | Removes an attribute on the help text element |
 | registerInputFocusHandler(handler: EventListener) => void | Registers an event listener on the native input element for a "focus" event |
-| deregisterInputFocusHandler(handler: EventListener) => void | Un-registers an event listener on the native input element for a "focus" event |
+| deregisterInputFocusHandler(handler: EventListener) => void | Deregisters an event listener on the native input element for a "focus" event |
 | registerInputBlurHandler(handler: EventListener) => void | Registers an event listener on the native input element for a "blur" event |
-| deregisterInputBlurHandler(handler: EventListener) => void | Un-registers an event listener on the native input element for a "blur" event |
+| deregisterInputBlurHandler(handler: EventListener) => void | Deregisters an event listener on the native input element for a "blur" event |
 | registerInputInputHandler(handler: EventListener) => void | Registers an event listener on the native input element for an "input" event |
-| deregisterInputInputHandler(handler: EventListener) => void | Un-registers an event listener on the native input element for an "input" event |
+| deregisterInputInputHandler(handler: EventListener) => void | Deregisters an event listener on the native input element for an "input" event |
 | registerInputKeydownHandler(handler: EventListener) => void | Registers an event listener on the native input element for a "keydown" event |
-| deregisterInputKeydownHandler(handler: EventListener) => void | Un-registers an event listener on the native input element for a "keydown" event |
+| deregisterInputKeydownHandler(handler: EventListener) => void | Deregisters an event listener on the native input element for a "keydown" event |
+| registerInputPointerDownHandler(evtType: string, handler: EventListener) => void | Registers an event listener on the native input element for a given event |
```

Comment:
I still think this API is confusing, as we have a number of other [de]registerInputFooHandler APIs which handle exactly 1 event type and do not receive an event type as input, whereas this _does_ receive an event type as input but is actually used for exactly 2 specific event types.

This seems like the kind of case where normally we would have a single pair of APIs,
 [de]registerInputInteractionHandler(evtType: string, handler: EventListener), and use it as needed for all event types for the input element. (The documentation for these APIs already sounds like exactly that, even.)

We can come back to this later, but if we're changing a lot of stuff for this component now, I figured we might want to actually clean up the adapter API while we're already making breaking changes.

Reply:
Alright. I'm doing it.

Reply:
I'm going to split it out in to an event handler for each element that needs one e.g.: `this.root_` will have one, and `this.input_` will as well. They take an event and an event handler and BOOM API surface shrunk by a third.